pycbc/inference/sampler_emcee.py

```diff
@@ -0,0 +1,993 @@
+            set this value to that size so that the array in the file will be
+            large enough to accomodate future data.
+        """
+        self.write_metadata(fp)
+        self.write_chain(fp, max_iterations=max_iterations)
+        self.write_likelihood_stats(fp, max_iterations=max_iterations)
+        self.write_acceptance_fraction(fp)
+
+    @staticmethod
+    def _read_fields(fp, fields_group, fields, array_class,
+            thin_start=None, thin_interval=None, thin_end=None, iteration=None,
+            temps=None, walkers=None, flatten=True):
+        """Base function for reading samples and likelihood stats. See
+        `read_samples` and `read_likelihood_stats` for details.
+
+        Parameters
+        -----------
+        fp : InferenceFile
+            An open file handler to read the samples from.
+        fields_group : str
+            The name of the group to retrieve the desired fields.
+        fields : list
+            The list of field names to retrieve. Must be names of groups in
+            `fp[fields_group/]`.
+        array_class : FieldArray or similar
+            The type of array to return. Must have a `from_kwargs` attribute.
+
+        For other details on keyword arguments, see `read_samples` and
+        `read_likelihood_stats`.
+
+        Returns
+        -------
+        array_class
+            An instance of the given array class populated with values
+            retrieved from the fields.
+        """
+        # walkers to load
+        if walkers is None:
+            walkers = range(fp.nwalkers)
+        if isinstance(walkers, int):
+            walkers = [walkers]
+
+        # temperatures to load
+        if temps is None:
+            temps = 0
+        if temps == 'all':
+            temps = range(fp.ntemps)
+        if isinstance(temps, int):
+            temps = [temps]
+
+        # get the slice to use
+        if iteration is not None:
+            get_index = iteration
+        else:
+            if thin_end is None:
+                # use the number of current iterations
+                thin_end = fp.niterations
+            get_index = get_slice(fp, thin_start=thin_start, thin_end=thin_end,
+                thin_interval=thin_interval)
+
+        # load
+        arrays = {}
+        group = fields_group + '/{name}/temp{tk}/walker{wi}'
+        for name in fields:
+            these_arrays = numpy.array([
+                    [fp[group.format(name=name, wi=wi, tk=tk)][get_index]
+                    for wi in walkers]
+                for tk in temps])
+            if flatten:
+                these_arrays = these_arrays.flatten()
+            arrays[name] = these_arrays
+        return array_class.from_kwargs(**arrays)
+
+    @classmethod
+    def read_samples(cls, fp, parameters,
+            thin_start=None, thin_interval=None, thin_end=None, iteration=None,
+            temps=0, walkers=None, flatten=True, samples_group=None,
+            array_class=None):
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->